addons/mail/models/mail_thread.py

```diff
@@ -694,6 +694,10 @@ def _notify_get_action_link(self, link_type, **kwargs):
             params['token'] = token
 
         link = '%s?%s' % (base_link, url_encode(params))
+
+        if self and hasattr(self, 'get_base_url'):
```

Comment:
Shouldn't that be correctly managed by /mail/ redirection controller instead of hardcoding strange stuff a bit everywhere in the code ?

Except from that, this is not modular and is likely to be broken by design.

Reply:
The redirection controller is already too late: if a user is logged in on a website (domain), but the link in the email is the base url (another domain), he will not be logged in there, so the redirect might not work as it should have.

Aside from that technical issue, the user is also not supposed to be aware there are multiple domains/websites in the first place. All links he receives should be from the domain that he knows and trusts, anything else would be confusing.

I think it's clear our system was originally not built to be used that way, but how do you suggest we solve this issue in a more modular way, especially for a fix in 12.0? In master, `get_base_url` has been moved to `BaseModel` already, so the `hasattr` condition here will become unnecessary.